tests/Fixtures/Test/FileReaderTest/StdinFakeStream.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixtures\Test\FileReaderTest;
+
+/**
+ * @author ntzm
+ *
+ * @internal
+ *
+ * @see \PhpCsFixer\Tests\FileReaderTest::testReadStdin
+ *
+ * A stream wrapper class that pretends to be php://stdin, and returns:
+ * - <?php echo "foo";
+ * - <?php echo "bar";
+ * alternating forever
+ */
+final class StdinFakeStream
+{
+    private $contents = array(
+        '<?php echo "foo";',
+        '<?php echo "bar";',
+    );
+
+    private static $contentsIndex = 0;
+
+    private $hasRead = false;
+
+    public function stream_open($path)
```

Comment:
it's following some magical interface, isn't it?

Reply:
Yep, magical: you must follow [streamWrapper class prototype](https://secure.php.net/manual/en/class.streamwrapper.php).

Reply:
maybe nice to link it in the class